### DISCUSSION TITLE
Upload new File via REST with specific ID

### DISCUSSION THREAD

**Discussion by MentalGear** [directus/directus#16763] [Archive]
I find the directus cloud docs a bit lacking. Is it possible to upload files directly to the file library (if so, is there a curl?), or only to a field of a collection record?

**Comment by br41nslug** [comment]
https://docs.directus.io/reference/files.html#upload-a-file

**Reply by MentalGear** [reply]
Thx, I assume it's not possible to upload a new file with a specific UUID (not update, but a new one) ?

**Reply by MentalGear** [reply]
Never mind, adding a form part "id" with a valid UUID BEFORE the "file" part works !

**Comment by MentalGear** [comment]
Found the endpoint, it was under the header "Returns" of https://docs.directus.io/reference/files.html#upload-a-file: /files paired with a multipart/form-data part with the name 'id'.

To specify an "id" with a valid UUID (or other props for the file) add the "part/property" BEFORE the "file" part !
